### DISCUSSION TITLE
Suppress default bibliography when using BibLaTeX and \printbibliography (Academic CV)

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#2431] [Q&A]
I'm trying to do an Academic CV, where it's necessary to print types (categories) of articles. To do this, I'm using BibLaTeX, `\printbibliography`, e.g., 

```latex
## Journal articles (peer reviewed)
\printbibliography[heading=none, type=article]

## Conference papers (peer reviewed)
\printbibliography[heading=none, type=inproceedings]
```

The problem is, I still get a **References** section at the end of my document (I guess it's by default).

I've tried `suppress-bibliography: true` in the metadata, but it doesn't appear to change anything. Perhaps because I'm using biblatex rather than citeproc?

How can I suppress that last part?

**Comment by cderv** [comment]
How are you setting the `\printbibliography` currently ? 

Unfortunately, it does not seem that Pandoc's template allows to pass option to `\printbibliography` easily - this would be a feature request to do in Pandoc IMO. Only `biblio-title` can be changed
https://github.com/jgm/pandoc/blob/eb1826740711319eccade8a634c03d747f461930/data/templates/default.latex#L529
This is the way to specific the title for the Reference section I believe. 

Our partial mechanism allows you to tweak the default Pandoc template though, among them `biblio.tex`
https://quarto.org/docs/journals/templates.html#latex-partials

This partials creates the bibliography, and you can tweak the call to `\printbibliography` there. You can use template with a document (https://quarto.org/docs/journals/templates.html#template-partials) or through an extensions for your Academic CV (https://quarto.org/docs/extensions/formats.html). See our template repo for custom LaTeX format (https://github.com/quarto-journals/article-format-template), to help create a format. 


`suppress-bibliography` is I believe only working when you use pandoc's own citation engine with CSL (https://quarto.org/docs/authoring/footnotes-and-citations.html#bibliography-generation) and not `natbib` or `bibtex`. 
When using Pandoc's citeproc, `reference-section-title` is the way to change the title. 

> The problem is, I still get a References section at the end of my document (I guess it's by default).

Do check that you don't have a `# References` at the end of your `qmd` document. Some examples do have it as it is useful with default Pandoc's citeproc, but not with other citation engine for LaTeX. 

Hope it helps

**Reply by fuhrmanator** [reply]
Thanks for all those useful pointers. Here's a follow-up:

https://github.com/quarto-dev/quarto-cli/blob/86f703c2d819a2a32bfe8eae9861b1537a2b7fa7/src/resources/formats/pdf/pandoc/biblio.tex#L28 is why the default **References** is printing. The other parts (see my question above) are just hard-coded latex `\printbibliography` inside the markdown. Since I have indeed activated `biblatex`, I would have to change that partial to prevent that default. It's good to know how to do it and I think it would be reasonable to do that.

In the short term, I simply turned off all the metadata flags for biblatex, and put everything in my own `preamble.tex` and use `include-in-header: preamble.tex`. It's a big short-term hack. Actually, I'm also using some [highlighting of names in the bibliography](http://www.hansenlab.org/cv_bibliography_tex) that required further hacking. I could make it all work with the partials mechanism of Quarto, but I don't have time for it now. 

Perhaps a flag like `suppress-biblatex-bibliography` to apply to the Quarto biblio.tex partial could be useful.

**Reply by cderv** [reply]
> Since I have indeed activated biblatex, I would have to change that partial to prevent that default. 

Changing `biblio-title` is not enough I supposed ? Just curious as I don't know as much about biblatex. 

> Perhaps a flag like suppress-biblatex-bibliography to apply to the Quarto biblio.tex partial could be useful.

I am not sure to follow. What would that do exactly ? If you want to remove the all `biblio.tex` content, you can pass an empty file as partial
````
format:
  pdf:
    template-partials:
      - biblio.tex # empty file
`````

> In the short term, I simply turned off all the metadata flags for biblatex,

What are the flags turned off ? `biblatex: false` ? 

And thanks for sharing the article, always cool to learn new things

**Reply by mcanouil** [reply]
Probably of interest because a related question was raised about using `biblatex-chicago`, see #2117.

**Reply by fuhrmanator** [reply]
> Changing biblio-title is not enough I supposed ? Just curious as I don't know as much about biblatex.

Not if you want to have titles for each section of an academic CV (Journal articles, Conference papers, etc. -- again see the sample markdown of my question). As far as I understand it, `biblio-title` would only change the title of one. Or am I misunderstanding how it can be used?

> If you want to remove the all biblio.tex content, you can pass an empty file as partial

True, but it's many steps (creating the blank file, supplying the metadata). But maintaining an option is also lots of work, so maybe the best compromise is to document how to achieve it.

> What are the flags turned off ? `biblatex: false` ?

No, I simply meant I didn't turn anything _on_ (there is absolutely no metadata specified relating to citations, not even specifying the `.bib` file). Everything is done in the preamble, as in pure LaTeX (bypassing Quarto's support for it). Since the biblio.tex partial has logic to look for that variable in metadata, not defining the variable serves the purpose of not executing the default `\printbibliography`.

**Reply by cderv** [reply]
Thanks a lot for your addition on all this.  that is helping me to understand. 

What I get from this:  We need to offer a CV template that covers also this usage of a reference by section. 
CV template is something already on backlog but not planned yet (https://github.com/quarto-dev/quarto-cli/discussions/705) 

I believe a specific CV format would be the best to support this usage. 

Thanks again for your feedback !
